File: cs/example/Simulator.cs
/*
 * File: Simulator.cs
 * Project: example
 * Created Date: 21/07/2021
 * Author: Shun Suzuki
 * -----
 * Last Modified: 09/10/2022
 * Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
 * -----
 * Copyright (c) 2022 Shun Suzuki. All rights reserved.
 * 
 */


using AUTD3Sharp;
using System;
using AUTD3Sharp.Utils;
using example.Test;

namespace example
{
    internal static class SimulatorTest
    {
        public static void Test()
        {
            Console.WriteLine("Test with Simulator");

            var autd = new Controller();
            autd.AddDevice(Vector3d.Zero, Vector3d.Zero);

            var link = new Link.Simulator().Port(50632).Build();
            if (!autd.Open(link))
            {
                Console.WriteLine(Controller.LastError);
                return;
            }

            TestRunner.Run(autd);
        }
    }
}

File: cpp/tests/link/simulator.cpp
// File: simulator.cpp
// Project: link
// Created Date: 26/09/2023
// Author: Shun Suzuki
// -----
// Last Modified: 27/09/2023
// Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
// -----
// Copyright (c) 2023 Shun Suzuki. All rights reserved.
//

#include <gtest/gtest.h>

#include <autd3/internal/controller.hpp>
#include <autd3/link/simulator.hpp>

TEST(Link, Simulator) {
  auto link = autd3::link::Simulator(8080).with_server_ip("127.0.0.1").with_timeout(std::chrono::milliseconds(200));

#ifdef RUN_LINK_SIMULATOR
  auto autd = autd3::internal::Controller::builder()
                  .add_device(autd3::internal::AUTD3(autd3::internal::Vector3::Zero(), autd3::internal::Vector3::Zero()))
                  .open_with(std::move(link));

  autd.close();
#else
  (void)link;
#endif
}
